x/leverage/types/errors.go

```diff
@@ -37,6 +37,7 @@ var (
 	ErrLiquidationIneligible = sdkerrors.Register(ModuleName, 403, "borrower not eligible for liquidation")
 	ErrMaxWithdrawZero       = sdkerrors.Register(ModuleName, 404, "max withdraw amount was zero")
 	ErrNoHistoricMedians     = sdkerrors.Register(ModuleName, 405, "insufficient historic medians available")
+	ErrMaxBorrowZero         = sdkerrors.Register(ModuleName, 406, "max borrow amount was zero")
```

Comment:
Why this is an error? If there is nothing more to borrow, we just return zero, rather error. Imaging smart contract just want to automatically set max borrow. It should handle zero value rather than dealing with failed transaction.

Reply:
Any supply / borrow / etc that is a _**No-Op**_ currently fails rather then succeeding with consumed gas. This follows suit

Reply:
If transaction fails, then the whole provided fee for gas is consumed anyway. So, it only makes doing composability more difficult.